### PR TITLE
fix English language typos

### LINKED ISSUE #3123: English Language Typos in Comments, Docs and Terminal Output
Running the `misspell` command (`go get -u github.com/client9/misspell/cmd/misspell`) identifies hundreds of English language typos (e.g., "ocurred" instead of "occurred") across the Traffic Control project. These are easily corrected and non-impacting. PR forthcoming.

### PR DESCRIPTION
**Pull request by jranson**
#### What does this PR do?

Fixes #3123 

#### Which TC components are affected by this PR?

- [x] Documentation
- [x] Grove
- [x] Traffic Analytics
- [x] Traffic Monitor
- [x] Traffic Ops
- [x] Traffic Ops ORT
- [x] Traffic Portal
- [x] Traffic Router
- [x] Traffic Stats
- [x] Traffic Vault
- [ ] Other _________

#### What is the best way to verify this PR?
These are documentation and other human/machine-consumable textual changes, so verification should be simple. There are no functional changes in this PR.

#### Check all that apply

- [ ] This PR includes tests
- [x] This PR includes documentation updates
- [ ] This PR includes an update to CHANGELOG.md
- [x] This PR includes all required license headers
- [ ] This PR includes a database migration (ensure that migration sequence is correct)
- [ ] This PR fixes a serious security flaw. Read more: [www.apache.org/security](http://www.apache.org/security/)

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: traffic_ops/app/lib/API/Tenant.pm [hunk 1]
```diff
@@ -138,21 +138,21 @@ sub update {
 			#Parent tenant to be set is not under user's tenancy
 			return $self->alert("Invalid parent tenant. This tenant is not available to you for parent assignment.");
 		}
-		my $parent_depth = $tenant_utils->get_tenant_heirarchy_depth($tenants_data, $params->{parentId});
+		my $parent_depth = $tenant_utils->get_tenant_hierarchy_depth($tenants_data, $params->{parentId});
```
Comment (ocket8888) [inline traffic_ops/app/lib/API/Tenant.pm]: Did you check to make sure that these function name changes didn't break anything?

File: traffic_ops/traffic_ops_golang/vendor/github.com/miekg/dns/edns.go [hunk 2]
```diff
@@ -490,7 +490,7 @@ func (e *EDNS0_N3U) String() string {
 	return s
 }
 
-// EDNS0_EXPIRE implementes the EDNS0 option as described in RFC 7314.
+// EDNS0_EXPIRE implements the EDNS0 option as described in RFC 7314.
```
Comment (ocket8888) [inline traffic_ops/traffic_ops_golang/vendor/github.com/miekg/dns/edns.go]: Should we be editing vendored dependencies?

Comment (dneuman64) [reply traffic_ops/traffic_ops_golang/vendor/github.com/miekg/dns/edns.go]: Probably not, @jranson can you revert this one?

Comment (dangogh) [reply traffic_ops/traffic_ops_golang/vendor/github.com/miekg/dns/edns.go]: definitely not..   that should be changed in the upstream dependency

File: traffic_ops/app/lib/API/Tenant.pm (hunk 1 above)
Comment (jranson) [reply traffic_ops/app/lib/API/Tenant.pm]: The corresponding function definitions in the Utils Module were updated along with the call updates here in the API Module. (See traffic_ops/app/lib/Utils/Tenant.pm down below). Thanks for checking!

File: traffic_ops/traffic_ops_golang/vendor/github.com/miekg/dns/edns.go (hunk 2 above)
Comment (jranson) [reply traffic_ops/traffic_ops_golang/vendor/github.com/miekg/dns/edns.go]: Thanks for catching, as I missed all of those deep vendor directories. They are reverted back. I'll submit PRs to the upstream projects for those typos.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2953/
Test PASSed.

File: traffic_ops/app/lib/API/Tenant.pm (hunk 1 above)
Comment (ocket8888) [reply traffic_ops/app/lib/API/Tenant.pm]: Alright, well I've got no problem with it if nobody else does

File: traffic_ops/app/lib/UI/Topology.pm [hunk 3]
```diff
@@ -914,7 +914,7 @@ sub compare_lists {
 
     my $list_compare_obj = List::Compare->new( \@$list_db, \@$list_disk );
 
-    my @db_only = $list_compare_obj->get_Lonly;
```
Comment (dangogh) [inline traffic_ops/app/lib/UI/Topology.pm]: I think this really *was* intended to be "Lonly"

**Comment by jranson** [comment]
My apologies on letting that one slip through!

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2982/
Test PASSed.

**Comment by dangogh** [comment]
cool..    can you also do `git rebase -i upstream/master` rather than merge?   the history will come out a lot cleaner. This all could be reduced to a single commit,  too..

**Comment by dangogh** [comment]
I know -- still makes the history messy, esp with "Merged from upstream/master" in there..   Just trying to encourage good practice...

**Comment by dneuman64** [comment]
oh right, I agree with that.  Rebase is much nicer.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3098/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3099/
Test PASSed.

**Comment by jranson** [comment]
As this one has been sitting around for some time without any action and the merge conflicts continue to pile up,  I'll just close it out.

**Pull request closed without merging**